### PR TITLE
feature/ADH-5307 highlight smart rules

### PR DESCRIPTION
**Pull request by KuzminIA**
task: https://tracker.yandex.ru/ADH-5307

### CODE REVIEW

File: smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss [hunk 1]
```diff
@@ -0,0 +1,68 @@
+$syntaxLiteral: #ff7f00;
```
Comment (remizov-arena) [inline smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss:1]: use css variables, pls

File: smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  &>.symbols\.ssmrule, &>.or\.ssmrule, &>.and\.ssmrule {
+    color: $syntaxSymbols;
+  }
+
+  &>.unrecognized\.ssmrule, &>.error\.ssmrule {
+    color: $syntaxError;
+  }
+
+  &>.every\.ssmrule, &>.at\.ssmrule, &>.from\.ssmrule, &>.to\.ssmrule, &>.now\.ssmrule {
```
Comment (remizov-arena) [inline smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss]: can we use `:is()` ?

File: smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.tsx
Comment (remizov-arena) [inline smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.tsx:1]: run `mvn com.mycila:license-maven-plugin:format -DskipTests` again, pls

File: smart-frontend/app/src/components/pages/RulesPage/RulesDialogs/RuleCreateDialog.tsx [hunk 3]
```diff
@@ -15,45 +15,39 @@
...
-  };
+  const handleChange = useCallback((value: string) => {
+    ruleText.current = value;
+  }, []);
 
   const handleCreate = () => {
     dispatch(createRuleWithUpdate(ruleText));
   };
 
-  const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
```
Comment (remizov-arena) [inline smart-frontend/app/src/components/pages/RulesPage/RulesDialogs/RuleCreateDialog.tsx:50]: can we try handle hot key for fast save?

Comment (KuzminIA) [reply smart-frontend/app/src/components/pages/RulesPage/RulesDialogs/RuleCreateDialog.tsx:50]: fixed, thank you

File: smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss (hunk 1 above)
Comment (KuzminIA) [reply smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss:1]: monaco doesn't support css variables
https://github.com/microsoft/monaco-editor/issues/2427

File: smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss (hunk 2 above)
Comment (KuzminIA) [reply smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.module.scss]: fixed, thank you

File: smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.tsx
Comment (KuzminIA) [reply smart-frontend/app/src/components/uikit/SmartRuleHighlighter/SmartRuleHighlighter.tsx:1]: fixed, thank you

**Review by remizov-arena — APPROVED** [review]

**Pull request merged**